### PR TITLE
Refactor/robot motion observer

### PR DESCRIPTION
**Pull request by martiniil**
A part of the integrationtest utility functions is replaced by the new test utility class `RobotMotionObserver`. A `RobotMotionObserver` object can detect, if the robot is moving and wait for a movement to start or to stop.

### CODE REVIEW

File: pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+class RobotMotionObserver(object):
+    """Observes motions of the robot.
+
+    The implementation is based on using the get_current_joint_values()) functionality of the RobotCommander().
+
+    :param group_name: Name of the planning group, default value is 'manipulator'
+    """
+
+    _DEFAULT_WAIT_TIME_FOR_MOTION_DETECTION = 3.0
+    _DEFAULT_TOLERANCE_FOR_MOTION_DETECTION = 0.01
```
Comment (agutenkunst) [inline pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py]: I think the unit is missing.

File: pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            curr_joint_values = self._get_current_joint_states()
+
+            if RobotMotionObserver._detect_motion(curr_joint_values, old_joint_values, move_tolerance):
+                motion_started = True
+                rospy.loginfo("Changed joint values detected: " + str(curr_joint_values))
+                rospy.loginfo("Motion started.")
+                break
+
+            if (rospy.get_time() - start_time > wait_time_out):
+                rospy.loginfo("Reached timeout when waiting for motion start.")
```
Comment (agutenkunst) [inline pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py:89]: To be really precise you should set `motion_started = False` to (better) cover edge cases.

File: pilz_robot_programming/test/integrationtests/tst_api_gripper.py [hunk 3]
```diff
@@ -33,6 +34,7 @@ class TestAPIGripper(unittest.TestCase):
 
     def setUp(self):
         self.robot = Robot(API_VERSION)
+        self.robot_motion_observer = RobotMotionObserver("gripper")
```
Comment (agutenkunst) [inline pilz_robot_programming/test/integrationtests/tst_api_gripper.py]: You could work with `RobotMotionObserver(group_name = "gripper") to be move expressiv.

**Review by agutenkunst — CHANGES_REQUESTED** [review]
Just minor change requests...

File: pilz_robot_programming/test/integrationtests/tst_api_gripper.py (hunk 3 above)
Comment (martiniil) [reply pilz_robot_programming/test/integrationtests/tst_api_gripper.py]: done.

File: pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py (hunk 2 above)
Comment (martiniil) [reply pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py:89]: not necessary.

File: pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py (hunk 1 above)
Comment (martiniil) [reply pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py]: done.

**Review by agutenkunst — APPROVED** [review]
